src/control-center/src/ControlCenter.cpp

```diff
@@ -6,13 +6,18 @@
 #include <blocks/BlockReader.h>
 #include <blocks/RequestConfigBlock.h>
 
-ControlCenter::ControlCenter(Serializer serializer) : serializer(serializer) {
+ControlCenter::ControlCenter(Serializer serializer, const std::string& filepath) : serializer(serializer) {
+    cfg = new ConfigReader(filepath);
+    port = cfg->read_integer(CC_PORT_PATH);
+    ip = cfg->read_string(CC_IP_PATH);
     connection.open_socket();
     con_send.open_socket();
     init_connection();
 }
 
-ControlCenter::~ControlCenter() {}
+ControlCenter::~ControlCenter() {
+    delete cfg;
```

Comment:
W nowoczesnym C++ staramy się pisać tak, by nie było w kodzie żadnego `delete`. Tutaj można to trywialnie zamienić na `unique_ptr` (do pary z `make_unique`).